pkg/manager/config/feature_gates_shim.go

```diff
@@ -0,0 +1,41 @@
+package config
+
+import (
+	"github.com/samber/lo"
+	"github.com/spf13/pflag"
+	"k8s.io/component-base/cli/flag"
+)
+
+// NewMapStringBoolForFeatureGatesWithDefaults takes a pointer to a FeatureGates (map[string]bool) and returns the
+// MapStringBool flag parsing shim for that map which populates it with defaults feature gates in case of missing keys.
+func NewMapStringBoolForFeatureGatesWithDefaults(m *FeatureGates) *FeatureGatesVar {
+	*m = lo.Must(newFeatureGates(nil)) // For nil it never returns an error.
+	return &FeatureGatesVar{fg: m}
+}
+
+type FeatureGatesVar struct {
```

Comment:
I think this should be internal as it's CLI-specific. There's a place for this kind of extension in `internal/manager/flags`.